### ISSUE TITLE
Make the project contributor-friendly

### ISSUE THREAD

**Issue by LPeter1997** [LanguageDev/Yoakke#38]
I've discussed this with a few people and I was told that we are not utilizing nearly enough features of GitHub to provide the most pleasant experience for new contributors. Here's a list of the bare minimum we should achieve:
 * [ ] Check out what the Insights->Community tab has to offer
   * [ ] Set up `Contributing.md`
 * [ ] Add issue templates to `.github/ISSUE_TEMPLATE`. [More info here.](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#configuring-the-template-chooser)
 * [ ] Enable Discussions in the repository
 * [ ] Organize our roadmap into "Milestones" or "Projects"
 * [ ] Start releasing packages on NuGet
   * [ ] Set up a stable release pipeline
   * [ ] Set up a nightly build pipeline. It might not seem important for such a young project, but I believe it helps to push out 
   * [About releasing NuGet packages](https://docs.microsoft.com/en-us/nuget/nuget-org/publish-a-package)
 * [ ] Use [docfx](https://github.com/dotnet/docfx) to generate a website for documentation

**Comment by skneko** [comment]
A note about NuGet packages. On commit [173150e](https://github.com/LanguageDev/Yoakke/commit/173150e60a8baf0fc36f839135da9cd9827ac732) the namespace `Yoakke.Ast` was renamed to `Yoakke.SyntaxTree`, but the associated NuGet packages published are still called [Yoakke.Ast](https://www.nuget.org/packages/Yoakke.Ast) and [Yoakke.Ast.Generator](https://www.nuget.org/packages/Yoakke.Ast.Generator). 

I don't know if NuGet allows package renaming, but it is very likely on the next desployment of NuGet packages (which will happen surely if we go ahead with the Nightly pipeline) we will need to do the following:
- Publishing new packages `Yoakke.SyntaxTree` and `Yoakke.SyntaxTree.Generator` with the aforementioned namespaces. This *may* require changing the project properties related to NuGet packages.
- Unpublishing and/or deprecating packages `Yoakke.Ast` and `Yoakke.Ast.Generator`.

--- 

On an unrelated note, GitHub allows for built-in, quick deployment of NuGet packages in the GitHub Packages registry, and has a ["packages" section](https://github.com/orgs/LanguageDev/packages?repo_name=Yoakke) in the repository overview and all. It has pricing but the free tier seems enough for this project. Maybe we could benefit from this feature. More info: https://github.com/features/packages

**Comment by LPeter1997** [comment]
Good point on the renaming, we will have to look for a solution there. We could just deprecate the packages, I'm fine with that.

I'll look into the GitHub Packages solution, thank you very much for the suggestion!
